git script: correct erased branch substitution

quetzai opened the pull request:
During the extraction of a job configuration, the substitution of variables
is done before git script starts looking for existing branches in the git repo.
If the template has already a branch or a commit specified, it is automatically
erased without checking.

This commit checks if such a treeish exists and avoid erasing it.

Commit:
git script: correct erased branch substitution

During the extraction of a job configuration, the substitution of variables
is done before git script starts looking for existing branches in the git repo.
If the template has already a branch or a commit specified, it is automatically
erased without checking.

This commit checks if such a treeish exists and avoid erasing it.

gvalkov commented on jenkins_autojobs/git.py:
```diff
@@ -96,15 +96,18 @@ def create_job(ref, template, config, ref_config):
+    # If el.text is not empty, the template has already had a treeish specified
+    # do not erase it
+    if not el.text:
+	# :todo: jenkins is being very capricious about the branch-spec
```
It can't be seen in the github diff viewer, but these lines start with tabs instead of spaces. This would result in a `TabError` in Python 3.

gvalkov commented:
Hi and thanks for looking into this. Accepting this PR would break the expectations of everyone who assumes the old behavior (i.e. always update the `branchSpec/name`).

I can think of two alternative solutions:

1.  Add an `ignore-jobs` config key, which would list job names that jenkins-autojobs should not touch. 

2. Make jenkins-autojobs ignore jobs with `(jenkins-autojobs-ignore)` in their description.

These have the added benefit of working for all SCM plugins.

Commit:
jenkins-autojobs: Add a 'ignore-jobs' feature

Introduction of a new feature to avoid erasing the value of the field
"branches to build".  So if the jenkins template has this field, it is
imported and not altered

Correct git.py to use this feature.

quetzai commented:
Hi,

I'm not sure this new commit is what you had in mind.
Since I'm far from an expert in svn and mercurial, I didn't touch these (not sure this feature would be usefull for svn).

Regards
